src/editors/containers/VideoEditor/components/VideoSettingsModal/components/TranscriptWidget/index.jsx

```diff
@@ -0,0 +1,145 @@
+
+/**
+ * Collapsible Form widget controlling video transcripts
+ */
+export const TranscriptWidget = ({
+  error,
+  // redux
+  transcripts,
+  allowTranscriptDownloads,
+  showTranscriptByDefault,
+  updateField,
+}) => {
+  const languagesArr = hooks.transcriptLanguages(transcripts);
+  const input = {
+    error: {
+      dismiss: () => { console.log('dismiss'); },
+      show: true,
+    },
+  };
+  const upload = {
+    error: {
+      dismiss: () => { console.log('dismiss'); },
+      show: true,
+    },
+  };
+
+  return (
+    <CollapsibleFormWidget
+      isError={Object.keys(error).length !== 0}
+      subtitle={languagesArr}
+      title="Transcript"
+    >
+      <ErrorAlert
+        dismissError={upload.error.dismiss}
+        hideHeading
+        isError={upload.error.show}
+      >
+        <FormattedMessage {...messages.uploadTranscriptError} />
+      </ErrorAlert>
+      <ErrorAlert
+        dismissError={input.error.dismiss}
+        hideHeading
+        isError={input.error.show}
+      >
+        <FormattedMessage {...messages.fileSizeError} />
+      </ErrorAlert>
+      <Stack gap={3}>
+        {transcripts ? (
+          <Form.Group className="mt-4.5">
+            <b>Transcript widget:</b>
+            <div className="mb-1">
+              <Form.Checkbox
+                checked={allowTranscriptDownloads}
+                className="mt-4.5 decorative-control-label"
+                onChange={(e) => updateField({ allowTranscriptDownloads: e.target.checked })}
+              >
+                <Form.Label>
+                  <FormattedMessage {...messages.allowDownloadCheckboxLabel} />
+                </Form.Label>
+              </Form.Checkbox>
+              <OverlayTrigger
+                key="right"
+                placement="right"
+                overlay={(
+                  <Tooltip>
+                    <FormattedMessage {...messages.tooltipMessage} />
+                  </Tooltip>
+                )}
+              >
+                <Icon className="d-inline-block mx-3" src={Info} />
+              </OverlayTrigger>
+            </div>
+            <Form.Checkbox
+              checked={showTranscriptByDefault}
+              className="mt-4.5 decorative-control-label"
+              onChange={(e) => updateField({ showTranscriptByDefault: e.target.checked })}
+            >
+              <Form.Label size="sm">
+                <FormattedMessage {...messages.showByDefaultCheckboxLabel} />
+              </Form.Label>
+            </Form.Checkbox>
+          </Form.Group>
+        ) : (
+          <>
+            <Alert variant="danger" icon={Info}>
+              Only SRT files can be uploaded. Please select a file ending in .srt to upload.
+            </Alert>
+            <FormattedMessage {...messages.addFirstTranscript} />
+          </>
+        )}
+        <Button iconBefore={FileUpload} onClick={() => { console.log('adding file'); }} variant="link">
```

Comment:
You'll want this to be a paragon Dropzone I think.

Reply:
Or at least be a button that opens up the file select object for now

Reply:
Might be better to use an invisible input + button in this case like this: https://bobbyhadz.com/blog/react-open-file-input-on-button-click